backend/src/common/plugins/file-upload/file-upload.plugin.ts

```diff
@@ -0,0 +1,50 @@
+import { type FastifyRequest } from 'fastify';
+import multer from 'fastify-multer';
+import {
+    type File as MulterFile,
+    type FileFilterCallback,
+} from 'fastify-multer/lib/interfaces.js';
+
+import {
+    AllowedExtensions,
+    AllowedMimeTypes,
+    FileSize,
+} from './enums/enums.js';
+
+const storage = multer.memoryStorage();
+
+const fileFilter = (
+    request: FastifyRequest,
+    file: MulterFile,
+    callback: FileFilterCallback,
+): void => {
+    if (
+        file.mimetype === AllowedMimeTypes.PDF ||
+        file.mimetype === AllowedMimeTypes.DOC ||
+        file.mimetype === AllowedMimeTypes.DOCX
```

Comment:
@MaksymGavatuk  Ok, but I have a question, should I always save the file in 'file' field or separate fieldnames for images and for resume docs for example
So that regarding to the fieldname I will check mimetypes and extensions.
```
Example:
1. Endpoint has preHandler: uploadFile.single('document'). Then check if fieldname 'document' - check for pdf,doc,docx
2. Endpoint has preHandler: uploadFile.single('image'). Then check if fieldname 'image' - check for jpg,png

```
Or should I just add additional checks to if() and that's all?

Reply:
@MaksymGavatuk I guess I'll do separation for fieldnames but the result always will be in the body.file

Reply:
You don't need to separate types